File: service_examples/src/service_server.cpp
#include<ros/ros.h>
#include<service_examples/custom_srv.h>
#include<iostream>


using namespace std;

bool callback(service_examples::custom_srvRequest &req,
              service_examples::custom_srvResponse &res)
{
    if (req.srvB.linear.x > 0)
    {
        res.srvC.data = true;
        cout<<"ok\n";
    }
    else 
    {
        res.srvC.data = false;
    }
    return true;
}


int main(int argc,char ** argv)
{
    ros::init(argc,argv,"service_server");
    ros::NodeHandle n;
    ros::ServiceServer server = n.advertiseService("speed_check",callback);
    ros::spin();

    return 0;
}

File: service_examples/src/service_server.py
#!/usr/bin/env python

import rospy
from service_examples.srv import custom_srv,custom_srvRequest,custom_srvResponse

def callBack(srv=custom_srvRequest()):
    result = custom_srvResponse()
    if srv.srvB.linear.x >0:
        rospy.loginfo('ok')
        result.srvC.data = True
    else :
         result.srvC.data = False

    return result


def main():
    server = rospy.Service('speed_check',custom_srv,callBack)
    rospy.spin()




if __name__ == '__main__':
    rospy.init_node('service_server')
    try :
        main()
    except rospy.ROSException() as e:
        print(e)
